lib/aws/s3svc/s3svc.go

```diff
@@ -0,0 +1,107 @@
+package s3svc
+
+import (
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+)
+
+// API represents a subset of the s3iface.
+type API interface {
+	DeleteObjects(*s3.DeleteObjectsInput) (*s3.DeleteObjectsOutput, error)
+	ListObjectVersions(*s3.ListObjectVersionsInput) (*s3.ListObjectVersionsOutput, error)
+}
+
+// Client provides a wrapper for s3api calls.
+type Client struct {
+	s3api   API
+	verbose bool
+}
+
+// NewClient returns a client struct that provides an interface to the s3 api.
+func NewClient(s3 API, v bool) *Client {
+	return &Client{
```

Comment:
There are no guards on the parameters. If they can truly be their default values, it'd be better to just make the fields on the struct exported and lose the factory function. Otherwise, add an err return value and guard the API.